### ISSUE TITLE
GHC 8 Support

### ISSUE THREAD

**Issue by creichert** [dmjio/stripe#43]
A new stackage nightly is ready for GHC 8 (`nightly-2016-05-26`). `stripe-haskell` was temporarily pulled from stackage for GHC 8 compatibility. I haven't taken a look at the specifics yet but it would certainly be nice to support this as quickly as possible.

Links for reference:
- https://github.com/fpco/stackage/blob/master/build-constraints.yaml#L19
- https://www.fpcomplete.com/blog/2016/05/moving-stackage-nightly-ghc-8

I'll try to follow up w/ exact compiler error messages.

**Comment by creichert** [comment]
The first round of errors I hit have to do with version bounds:

```
*** Resolver nightly-2016-05-26 will need external packages: 
    http-streams not found
        - stripe-http-streams requires >=0.7 && <1.0
    time version 1.6.0.1 found
        - stripe-core requires >=1.4 && <1.6
        - stripe-tests requires >=1.4 && <1.6
    transformers version 0.5.2.0 found
        - stripe-core requires >=0.3 && <0.5
        - stripe-tests requires >=0.3 && <0.5
```

I bumped the `stripe-core` and `stripe-tests` versions for `time` & `transformers` but didn't have any luck w/ `http-streams` yet.

**Comment by creichert** [comment]
I filed an issue for `http-streams` here: https://github.com/afcowie/http-streams/issues/96

I believe some of the necessary changes have been made in `snap-core` already:
snapframework/snap-core#255

**Comment by dmjio** [comment]
I'll try to fix this tonight

**Comment by dmjio** [comment]
https://github.com/dmjio/stripe/pull/44

**Comment by dmjio** [comment]
@creichert this should be done, pending afcowie/http-streams#96

**Comment by 3noch** [comment]
The blocking issue is now closed. Is this done?

**Comment by creichert** [comment]
I'm assuming this issue is fixed, yet stripe-haskell is still not available in stackage: https://github.com/fpco/stackage/blob/master/build-constraints.yaml#L69

**Comment by dmjio** [comment]
@creichert, I have a few other fixes to batch up, but I can try to push out a release this week to hackage, and stackage.

**Comment by creichert** [comment]
@dmjio That would be really awesome, thank you

**Comment by 3noch** [comment]
@dmjio Yah I need this too.

**Comment by creichert** [comment]
One quick note, it appears that stripe is blocked on aesone-1.0: https://github.com/fpco/stackage/blob/master/build-constraints.yaml#L71

I haven't yet had a chance to try tweaking version bounds but i'll speak up if I get there first :)

**Comment by jazzyeagle** [comment]
I was able to get the code to compile using GHC 8.0.2 (using Stack Nightly-2017-01-27) by modifying the constraints for hspec (<2.4) & and hspec-core (<2.4).  It built w/ version 2.3.2 for both.  Any way we can get this updated and then have stack updated to include stripe as part of the lts again?

Thanks!

**Comment by dmjio** [comment]
@jazzyeagle @creichert ghc80* is now supported, is there anything I need to do specifically to ensure it's on stackage?

**Comment by cdepillabout** [comment]
@dmjio I believe you have to send a PR that edits the stackage `build-constraints.yaml` file and make sure the stripe libraries are uncommented:

https://github.com/fpco/stackage/blob/master/build-constraints.yaml#L90

From what I've heard, the stackage maintainers will sometimes do this for you manually, but they don't do it on a regular basis, so you might be waiting a couple weeks.  The easiest way is to just send a PR.

Also, I'd be really grateful if the stripe library was in a future release of `lts-8` (it's currently not in `lts-8.0`).  Here's the steps for getting a package into an existing `lts` release:

https://github.com/fpco/stackage/blob/master/MAINTAINERS.md#adding-a-package-to-an-lts-snapshot

Basically it just requires opening an issue on the https://github.com/fpco/lts-haskell repo.

**Comment by dmjio** [comment]
@cdepillabout @creichert @jazzyeagle, `stripe-*` is now compatible with GHC8. Going to close this and make separate issue for LTS.
